src/enzyme_ad/jax/Passes/AffineToStableHLORaising.cpp

```diff
@@ -944,6 +978,217 @@ emitStoreAsScatter(Location loc, Value update, Value input, ValueRange sIndices,
   return res;
 }
 
+static LogicalResult tryRaisingForOpToStableHLOUnroll(
+    affine::AffineForOp forOp, IRMapping &mapping, OpBuilder &builder,
+    llvm::DenseMap<Value, affine::AffineValueMap> &maps, ParallelContext pc) {
+  auto tmpBlock = new Block();
+  OpBuilder oldFuncBuilder(builder.getContext());
+  oldFuncBuilder.setInsertionPointToStart(tmpBlock);
+  auto clonedFor = cast<affine::AffineForOp>(oldFuncBuilder.clone(*forOp));
+  auto yield = oldFuncBuilder.create<affine::AffineYieldOp>(
+      clonedFor.getLoc(), clonedFor.getResults());
+  if (failed(affine::loopUnrollFull(clonedFor))) {
+    delete tmpBlock;
+    return failure();
+  }
+  for (auto &innerOp : tmpBlock->without_terminator()) {
+
+    if (tryRaisingOpToStableHLO(&innerOp, mapping, builder, maps, pc).failed())
+      return failure();
```

Comment:
This leaks the block. Just put it in a `unique_ptr` so it's always deleted when exiting the scope and `.release` it when merging below.

Reply:
Oh yes unique_ptr is a great idea. I completely missed this leak....